client.go

```diff
@@ -174,101 +210,197 @@ func (c *Client) Init() {
 }
 
 func (c *Client) doV2PutRequest(endpoint string, req interface{}) ([]byte, error) {
-	return c.doRequest("PUT", endpoint, req, 2)
+	return c.doRequest(http.MethodPut, endpoint, req, 2)
 }
 
 func (c *Client) doV2PostRequest(endpoint string, req interface{}) ([]byte, error) {
-	return c.doRequest("POST", endpoint, req, 2)
+	return c.doRequest(http.MethodPost, endpoint, req, 2)
 }
 
 func (c *Client) doV2DeleteRequest(endpoint string, req interface{}) ([]byte, error) {
-	return c.doRequest("DELETE", endpoint, req, 2)
+	return c.doRequest(http.MethodDelete, endpoint, req, 2)
 }
 
-func (c *Client) doRequest(method, uri string, body interface{}, apiVersion int) ([]byte, error) {
+func (c *Client) doRequest(method, uri string, body interface{}, apiVersion int) (res []byte, err error) {
 	var bts []byte
 	if body != nil {
-		var err error
-		bts, err = json.Marshal(body)
-		if err != nil {
-			return nil, err
+		if bts, err = json.Marshal(body); err != nil {
+			return nil, fmt.Errorf("unable to marshal request body: %w", err)
 		}
 	}
 
 	var url string
 	switch apiVersion {
 	case 1:
-		url = endpoint(uri)
+		url = c.endpoint(uri)
 	case 2:
-		url = endpointV2(uri)
+		url = c.endpointV2(uri)
 	default:
 		return nil, fmt.Errorf("aiven API apiVersion `%d` is not supported", apiVersion)
 	}
 
-	retryCount := 2
-	for {
-		req, err := http.NewRequest(method, url, bytes.NewBuffer(bts))
-		if err != nil {
-			return nil, err
-		}
-
-		req.Header.Set("Content-Type", "application/json")
-		req.Header.Set("User-Agent", c.UserAgent)
-		req.Header.Set("Authorization", "aivenv1 "+c.APIKey)
+	req, err := retryhttp.NewRequest(method, url, bytes.NewBuffer(bts))
+	if err != nil {
+		return nil, fmt.Errorf("unable to build http request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", c.userAgent)
+	req.Header.Set("Authorization", "aivenv1 "+c.apiKey)
 
-		rsp, err := c.Client.Do(req)
-		if err != nil {
-			return nil, err
-		}
-		defer func() {
-			err := rsp.Body.Close()
-			if err != nil {
-				log.Printf("[WARNING] cannot close response body: %s \n", err)
-			}
-		}()
-
-		responseBody, err := ioutil.ReadAll(rsp.Body)
-		// Retry a few times in case of request timeout or server error for GET requests
-		if (rsp.StatusCode == 408 || rsp.StatusCode >= 500) && retryCount > 0 && method == "GET" {
-			retryCount--
-			continue
-		} else if rsp.StatusCode < 200 || rsp.StatusCode >= 300 {
-			return nil, Error{Message: string(responseBody), Status: rsp.StatusCode}
-		}
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("unable to perform http request: %w", err)
+	}
+	defer func() { _ = rsp.Body.Close() }()
 
-		return responseBody, err
+	res, err = ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read request body: %w", err)
+	}
+	switch sc := rsp.StatusCode; {
+	case sc >= 200 && sc < 300:
+		// 2xx
+		return res, nil
+	case sc >= 400 && sc < 600:
```

Comment:
We would need to transform this to the `aiven.Error` to make `aiven.IsNotFound`() and other error methods to work correctly.